### PR TITLE
ARROW-1300: [JAVA] Fix Tests for ListVector

### PR DESCRIPTION
**Pull request by siddharthteotia**
@StevenMPhillips 

Fixed the following:

(1) TestListVector.java doesn't include tests for nested lists where the underlying dataVector for a listVector is also a listVector.

(2) The copy test in TestListVector.java only checks the bit vector contents and doesn't verify the actual contents of list vector

### CODE REVIEW

**Comment by siddharthteotia** [comment]
For some reason, git shows more diff in changes than I have actually made.

I have added lines 42, 96-115 and 457-572 

however, git shows a lot of additional space/tab changes which I haven't really made

**Comment by wesm** [comment]
It looks like there's still a bunch of style issues in the codebase. Some time ago we added checkstyle checks but it's still showing a lot of working. It would be good to get checkstyle to a clean state cc @icexelloss

**Comment by siddharthteotia** [comment]
thanks @wesm 

In the latest commit, I have tried to get rid of most of the tab/space changes.

**Comment by icexelloss** [comment]
@wesm, I can help with clean up the style. Track here: https://issues.apache.org/jira/browse/ARROW-1304

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
 import org.junit.Test;
 
 import java.util.List;
+import java.util.ArrayList;
```
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Should these two import switch order? `sortImportsInGroupAlphabetically`

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java [hunk 2]
```diff
@@ -92,6 +93,26 @@ public void testCopyFrom() throws Exception {
       Assert.assertFalse("should be null", reader.isSet());
       reader.setPosition(2);
       Assert.assertTrue("shouldn't be null", reader.isSet());
+
+      /* check the exact contents of vector */
+      final ListVector.Accessor accessor = outVector.getAccessor();
+
+      /* index 0 */
+      Object result = accessor.getObject(0);
+      ArrayList<Long> resultSet = (ArrayList<Long>)result;
```
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Need space between `(ArrayList<Long>)` and `result`

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java [hunk 3]
```diff
@@ -415,15 +433,14 @@ public void testSplitAndTransfer() throws Exception {
             dataLength1 = offsetAccessor.get(start + i + 1) - offsetAccessor.get(start + i);
             dataLength2 = offsetAccessor1.get(i + 1) - offsetAccessor1.get(i);
 
-            assertEquals("Different data lengths at index: " + i + " and start: " + start,
-                         dataLength1, dataLength2);
+            assertEquals("Different data lengths at index: " + i + " and start: " + start, dataLength1, dataLength2);
```
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: This line is more than 100 length

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java (hunk 2 above)
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Actually, [google_checks.xml](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml) doesn't specify [TypecastParenPad](http://checkstyle.sourceforge.net/config_whitespace.html#TypecastParenPad) rule, so this is fine.

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java [hunk 4]
```diff
@@ -433,4 +450,120 @@ public void testSplitAndTransfer() throws Exception {
...
+      Object result = accessor.getObject(0);
+      ArrayList<ArrayList<Long>> resultSet = (ArrayList<ArrayList<Long>>)result;
+      ArrayList<Long> list;
+
+      assertEquals(2, resultSet.size());              /* 2 inner lists at index 0 */
+      assertEquals(3, resultSet.get(0).size());       /* size of first inner list */
+      assertEquals(4, resultSet.get(1).size());      /* size of second inner list */
+
+      list = resultSet.get(0);
+      assertEquals(new Long(50), (Long)list.get(0));
```
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Why do you need to cast to Long here?

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java [hunk 5]
```diff
@@ -433,4 +450,120 @@ public void testSplitAndTransfer() throws Exception {
...
+
+      assertEquals(2, listVector.getMutator().getLastSet());
+
+      listVector.getMutator().setValueCount(2);
+
+      final ListVector.Accessor accessor = listVector.getAccessor();
+      assertEquals(2, accessor.getValueCount());
+
+      /* get listVector value at index 0 -- the value itself is a listvector */
+      Object result = accessor.getObject(0);
```
Comment (icexelloss) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Unrelated:

Is there a way to get unboxed value from nested list vector?

**Comment by StevenMPhillips** [comment]
+1

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java (hunk 4 above)
Comment (siddharthteotia) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Not needed. Addressed

File: java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java (hunk 3 above)
Comment (siddharthteotia) [inline java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java]: Addressed

**Comment by siddharthteotia** [comment]
Addressed review comments in the latest commit.

**Comment by icexelloss** [comment]
+1

**Pull request closed without merging**